### ISSUE TITLE
[bug]Grafana/openTSDB 2.2 missing tag conditions from nested template variable in the query filters parameters

### ISSUE THREAD

**Issue by Helene** [grafana/grafana#8180]
- What Grafana version are you using?
4.2
- What datasource are you using?
OpenTSDB 2.2
- What OS are you running grafana on?
Windows 7

Hello,
I have the following filesystem and storage pools setup in a gpfs cluster:

Storage pools in file system 'fs1': >          system                  
Storage pools in file system 'fs2':  >         system ,             bobPool ,             myPool               
Storage pools in file system 'gpfs0':  >     system,              data                 
Storage pools in file system 'objfs':  >      system,              data      

As you can see all filesystems have  pool with name: 'system' (default pool created from system as soon as new filesystem is created). To include only pools from fs2 in the metric graph I created nested template variable($value2):   
![nestedtemplatingjpg](https://cloud.githubusercontent.com/assets/22845495/25248422/7ddc458e-260e-11e7-9b8b-c9886f1413bb.JPG)
and passed it to the metric query filter:
![query_withnestedtemplating](https://cloud.githubusercontent.com/assets/22845495/25248470/acb28b52-260e-11e7-9ad9-b8afdf3bc92f.JPG)
Unfortunately the graph shows the time series for all pools named 'system', even they don't belong to the filesystem fs2. Looking at the  /api/query POST request parameters I see that the filesystem template variable used to filter tag values in $value2 is not included in the query request parameters:
![inputquery](https://cloud.githubusercontent.com/assets/22845495/25249067/cb0b20e4-2610-11e7-9b43-68a5dc94884a.JPG)
I was expecting to see 2 filters passed to the query: { gpfs_fs_name=fs2, gpfs_diskpool_name=system|myPool|bobPool}

------------
I tried to get the expecting results with a "value groups/tags" template variable, but I got the same result:

![groupquery1](https://cloud.githubusercontent.com/assets/22845495/25249462/35a43c3c-2612-11e7-8dc5-c8a9f2758d45.JPG)

![valuegrouptempl](https://cloud.githubusercontent.com/assets/22845495/25249295/b2225c9a-2611-11e7-9752-22e06adafb35.JPG)

![grouptempqueryjpg](https://cloud.githubusercontent.com/assets/22845495/25249309/c1164c2a-2611-11e7-8cb4-c97947762fd4.JPG)

![params](https://cloud.githubusercontent.com/assets/22845495/25249327/d14fe2fe-2611-11e7-80d4-4667a78fd9e8.JPG)

**Comment by torkelo** [comment]
The query Grafana sends to OpenTSDB looks correct to me:
![image](https://cloud.githubusercontent.com/assets/10999/25264833/d69cf2c6-2669-11e7-92e7-dfdef2e18673.png)

See the filter property contains all you selected values `bobPool|myPool|system`

**Comment by torkelo** [comment]
> I was expecting to see 2 filters passed to the query: { gpfs_fs_name=fs2, gpfs_diskpool_name=system|myPool|bobPool}

I do no see this filter in your query.

**Comment by Helene** [comment]
This is exactly my problem. The template variable $value2 is defined as follows:
tag_values($metric, gpfs_diskpool_name, $tag=$value1).  With selected $metric=gpfs_diskpool_name, $tag=gpfs_fs_name,  $value1=fs2 the **_lookup request_** for $value2 looks like:

http://localhost:8099/api/datasources/proxy/1/api/search/lookup?limit=3000&m=gpfs_pool_free_dataKB{gpfs_diskpool_name=*,**gpfs_fs_name=fs2**} 

and returns **_correct values_**: 

![lookupjpg](https://cloud.githubusercontent.com/assets/22845495/25266824/e8497004-2672-11e7-8135-d1eb448e0ed4.JPG)

The same variable $value2 used in **_the graph query_**. I'm missing **gpfs_fs_name=fs2** passed to the query filters, since $value2 is nested variable and has additional filter ($tag=$value1) for filtering pools only from filesystem fs2.

![queryfilter](https://cloud.githubusercontent.com/assets/22845495/25267341/a5f1196c-2674-11e7-9266-d8e68e08cdcc.JPG)

This results in more then 3 time series shown in the Graph, since the pool with name 'system' exists also in filsystem fs1, gpfs0, objfs.

In other words the same variable $value2 used in /lookup and /query request returns different number of ts's in results.

**Comment by torkelo** [comment]
but you only use $value2 in your query, no filter for gpfs_fs_name exists in the metric query

**Comment by Helene** [comment]
Yes, because I select gpfs_fs_name using $value1 and $value1 is used in the **nested** variable $value2. It couldn't be that the _search/lookup request_ retrieves for $value2 = ...{gpfs_diskpool_name= "\*", gpfs_fs_name=fs2} and the _query request_  retrieves for the same variable $value2 =...{gpfs_diskpool_name="\*"}

**Comment by torkelo** [comment]
> Yes, because I select gpfs_fs_name using $value1 and $value1 is used in the nested variable $value2. It couldn't be that the search/lookup request retrieves for $value2 = ...{gpfs_diskpool_name= "*", gpfs_fs_name=fs2} and the query request retrieves for the same variable $value2 =...{gpfs_diskpool_name="*"}

Not sure what you mean. All I know is that the query includes only the value2 filter, and the values you specified for value2 is sent by Grafana in the query. There is no filter for gpfs_fs_name tag in the query. 

So if your getting a response that looks strange, maybe it's OpenTSDB that is returning something unexpected?

**Comment by torkelo** [comment]
So basically this:
````
gpfs_fs_name gpfs_diskpool_name
fs2 system
fs2 myPool
fs2 bobPool
fs1 system
fs1 otherPool
fs1 otherPool2

gpfs_diskpool_name tag exists in with two different gpfs_fs_name tag values (fs1, fs2), so if you only filter by gpfs_diskpool_name tag you will get series with different gpfs_fs_name, since you do not filter by this tag
```

**Comment by Helene** [comment]
you are correct with one small remark, for filtering I'm using the template variable $value2 which means give me tag_values for gpfs_diskpool_name having gpfs_fs_name=fs2 (the additional filesystem filter is already included in the varible $value. And this works for the LOOKUP request. I'm expecting it works for GET QUERY request also)

**Comment by torkelo** [comment]
but how your $value2 variable is configured does not matter, the templating system is a string interpolation system. $value2 is going to be interpolated based on it's current value and position in your metric query. 

How is grafana going to know how you want to use it? Do you expect Grafana to look at the definition and try to modify the query? That would go wrong so many times and would be insanely complicated

**Comment by Helene** [comment]
In my opinion all conditions defined in the variable should be passed to the query. Otherwise I see no reason to create such variable with additional filtering conditions.

**Comment by torkelo** [comment]
@Helene I think your are missunderstanding the templating system greatly. Not blaming you, probably bad docs. But what your asking for makes no sense. 

You variables with a current value (can be multiple for multi value variables), and you have options to choose from. You can can define queries to fill the options to choose from so they dynamic. You can link variables together so the options in one depend on the value of another. 
--------------------

You then use these variables in Panel titles and in metric query expressions. The variables are placeholders for the value you have selected. So if you specify filter like `gpfs_diskpool_name=$value2` grafana is going to interpolate the current value of `$value2` before sending the query. Since you have no gpfs_fs_name filter in your query no such filter is applied. Impossible for Grafana to know that you also want another filter automatically added to your query. This is not a templating system your asking for it's some magic AI :)
